src/gpio.c

```diff
@@ -0,0 +1,196 @@
+    }
+
+    close(fd);
+    return status;
+}
+
+int8_t gpio_unexport(uint8_t pin)
+{
+    int fd;
+    int8_t bytes_written, status = 0;
+    uint8_t buffer[BUFFER_MAX];
+
+    fd = open("/sys/class/gpio/unexport", O_WRONLY);
+    if (fd == -1)
+    {
+        fprintf(stderr, "Failed to open unexport for writing");
+        return -1;
+    }
+
+    bytes_written = snprintf(buffer, BUFFER_MAX, "%" PRIu8, pin);
+    status = write(fd, buffer, bytes_written);
+    if (status == -1)
+    {
+        fprintf(stderr, "Failed to unexport gpio");
+    }
+
+    close(fd);
+    return status;
+}
+
+int8_t gpio_direction(uint8_t pin, uint8_t direction)
+{
+    int fd;
+    int8_t status = 0;
+    char* directions_str[] = {"in", "out"};
+    char path[PATH_MAX];
+
+    snprintf(path, PATH_MAX, "/sys/class/gpio/gpio%" PRIu8 "/direction", pin);
+    fd = open(path, O_WRONLY);
+    if (fd == -1)
+    {
+        fprintf(stderr, "Failed to open gpio direction for writing\n");
+        return -1;
+    }
+
+    status = write(fd, directions_str[direction], strlen(directions_str[direction]));
+    if (status == -1)
+    {
+        fprintf(stderr, "Failed to set gpio direction\n");
+    }
+
+    close(fd);
+    return status;
+}
+
+int8_t gpio_edge(uint8_t pin, uint8_t type)
+{
+    int fd;
+    int8_t status = 0;
+    char* type_str[] = {"rising", "falling"};
+    char path[PATH_MAX];
+
+    snprintf(path, PATH_MAX, "/sys/class/gpio/gpio%" PRIu8 "/edge", pin);
+    fd = open(path, O_WRONLY);
+    if (fd == -1)
+    {
+        fprintf(stderr, "Failed to open gpio edge for writing\n");
+        return -1;
+    }
+
+    status = write(fd, type_str[type], strlen(type_str[type]));
+    if (status == -1)
+    {
+        fprintf(stderr, "Failed to set gpio edge type\n");
+    }
+
+    close(fd);
+    return status;
+}
+
+int8_t wait_for_interrupt(uint8_t gpio, uint8_t timeout)
+{
+    int fd;
+    int8_t status;
+    struct pollfd pfd;
+    char path[PATH_MAX];
+    char buffer[BUFFER_MAX] = {0};
+
+    sprintf(path, "/sys/class/gpio/gpio%" PRIu8 "/value", gpio);
+    if ((fd = open(path, O_RDONLY)) < 0)
+    {
+        perror("error: failed to open gpio\n");
+        return -1;
+    }
+
+    memset(&pfd, 0, sizeof(pfd));
+    pfd.fd = fd;
+    pfd.events = POLLPRI;
+
+    lseek(pfd.fd, 0 , SEEK_SET);
+    status = read(fd, buffer, sizeof(buffer));
+    if (status < 0)
+    {
+        perror("read()");
+        close(fd);
+        return -1;
+    }
+
+    while(running)
+    {
+        memset(buffer, 0x00, 8);
+        memset(&pfd, 0, sizeof(pfd));
+        pfd.fd = fd;
+        pfd.events = POLLPRI;
+
+        lseek(pfd.fd, 0 , SEEK_SET);
+
+        printf("Ready for detecting motion...\n");
+        status = poll(&pfd, 1, timeout * 1000);
+        if (status == -1)
+        {
+            perror("poll()");
+            close(fd);
+            return -1;
+        }
+        else if (status == 0)
+        {
+            printf("timeout\n");
```

Comment:
what's the point of timeout if it does nothing more than just print a timeout string?